bin/cml/comment/create.js

```diff
@@ -18,16 +18,25 @@ exports.builder = (yargs) =>
     .options(exports.options);
 
 exports.options = kebabcaseKeys({
+  target: {
+    type: 'string',
+    description:
+      'Forge object to create comment on, can be one of pr, commit or issue. ' +
+      "Specify 'issue#123' to create a comment on a specific issue. " +
+      'By default cml will create a PR comment if running in a forge PR-related action ' +
+      'or if HEAD is in a PR branch. Otherwise a commit comment will be created.'
```

Comment:
migrated feedback from https://github.com/iterative/cml/pull/1241#discussion_r1024437445 as requested:

I think we need:

- `#123` to auto-infer pr-or-issue
- `mr` alias for `pr`
- `mr#123` alias for `pr#123`
- why do we have `commit#d34db3`? Doesn't `--commit-sha=d34db3` already cover this?
- full URL (e.g. if they want to target another PR returned by `cml pr`)

Reply:
>     * `#123` to auto-infer pr-or-issue
Why? Is there any confusion among users whether they are adding a PR or issue comment?

>     * `mr` alias for `pr`
>     * `mr#123` alias for `pr#123`
Did we have a `--mr` alias for the `--pr` flag?

>     * full URL (e.g. if they want to target another PR returned by `cml pr`)
This could be done, but probably in a follow-up.

> Also, why do we have `commit#d34db3`? Doesn't `--commit-sha=d34db3` already cover this?
As @dacbd mentioned, this is the whole point of `--target` - to get rid of the `--pr`, `--commit-sha` flags (and not introduce `--issue`).

Reply:
ok fine by me to move this to a follow-up issue